### DISCUSSION TITLE
Un-deprecate `assembleLayout()`, or abstract `assembleLayoutFromPost()`

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/cms#9676] [Ideas] [answered]
Refer to https://github.com/craftcms/cms/issues/6625 I'd like to see the current `assembleLayoutFromPost()` main code abstracted out to another function to not rely on POST params. We've hit this a few times, with a few different scenarios.

[Field Manager](https://github.com/verbb/field-manager) in particular now has this issue, where it allows exporting field config to JSON, for later import.

As a demonstration of this issue, see https://github.com/spicywebau/craft-neo/pull/489 where the reliance on the field layout being assembled via POST params make it impossible to assign a field layout to a blocktype. Despite having the exact same config data, just from stored JSON. The PR linked scoots around this issue.

[Vizy](https://github.com/verbb/vizy) has similar issues, where the data for field settings, where the data is stored a bit differently to play nice with Vue, and multiple block type definitions. I'm sure this one is an edge-case of my own making.

Opening this as a discussion because:
- Probably need to chat about the best way to handle this, before a PR
- You'd be making this change to help plugin/module developers as its a non-core issue

**Comment by brandonkelly** [comment, marked as answer]
Craft already has a built-in way to serialize a field layout into a portable, JSON-encodable array ([FieldLayout::getConfig()](https://github.com/craftcms/cms/blob/87c6e788af4d463285d5cfd82f143596e4407ee3/src/models/FieldLayout.php#L439)), and a way to unserialize those arrays back into FieldLayout models ([FieldLayout::createFromConfig()](https://github.com/craftcms/cms/blob/87c6e788af4d463285d5cfd82f143596e4407ee3/src/models/FieldLayout.php#L133)). They’re both used for storing field layouts in the project config, but could apply to other use cases as well.

The main reason `assembleLayout()` has been deprecated is because it doesn’t support any of the new field layout features that were added in Craft 3.5. If it weren’t for that, I wouldn’t have any problem continuing to support it even if only for plugins’ sake.

**Comment by engram-design** [comment]
Yep, and that's exactly what I'm using `getConfig/createFromConfig` (thanks for that being super-helpful). I suppose the assembly of line items from POST, versus an object provided to the function is the issue, but I guess I'll keep handling this the way I have.

**Reply by brandonkelly** [reply]
Couldn’t you call `assembleLayoutFromPost()` when the data is in POST, then call `getConfig()` on the resulting FieldLayout model?

**Reply by engram-design** [reply]
I guess this is 1. More of a Neo issue that they need to allow for more than just creating field layouts from POST, and 2. Vizy just needs to be re-written to use `assembleLayoutFromPost()` and rely on POST data, instead of its own `fieldData` property that's already been populated when saving the field settings.

Happy to drop this.
